src/app/pages/system/CloudCredentials/cloudcredentials-form/cloudcredentials-form.component.ts

```diff
@@ -793,14 +797,59 @@ export class CloudCredentialsFormComponent {
         }
       ]
     },
+    // show if provider support oauth
+    {
+      type: 'input',
+      name: 'client_id',
+      placeholder: helptext.client_id.placeholder,
+      tooltip: helptext.client_id.tooltip,
+      isHidden: true,
+    },
+    {
+      type: 'input',
+      name: 'client_secret',
+      placeholder: helptext.client_secret.placeholder,
+      tooltip: helptext.client_secret.tooltip,
+      isHidden: true,
+    },
   ];
 
-
   protected providers: Array<any>;
   protected providerField: any;
   protected entityForm: any;
+  protected oauthClentIdField: any;
+  protected oauthClentSecretField: any;
 
   public custActions: Array<any> = [
+    {
+      id: 'authenticate',
+      name: T('Authenticate'),
+      function: () => {
+        window.open(this.oauthURL+ "?origin=" + encodeURIComponent(window.location.toString()), "_blank", "width=640,height=480");
```

Comment:
Why a fixed 640x480 window?  We really should scale dialogs and new browser windows to like 80% of the main browser window. It might be much larger (big monitor) or much smaller (phone) than a fixed number.

Reply:
Hi @wblock , this one is a new scalable window. it will show as a new tab on the phone.